packages/flutter/test/animation/animation_controller_test.dart

```diff
@@ -292,4 +292,30 @@ void main() {
     expect((){ controller.repeat(period: null); }, throwsFlutterError);
   });
 
+  test('Do not animate if already at target', () {
+    final List<AnimationStatus> statusLog = <AnimationStatus>[];
+
+    final AnimationController controller = new AnimationController(
+      value: 0.5,
+      vsync: const TestVSync(),
+    )..addStatusListener(statusLog.add);
+
+    controller.animateTo(0.5, duration: const Duration(milliseconds: 100),);
+    expect(statusLog, equals(<AnimationStatus>[ AnimationStatus.completed ]));
+    expect(controller.value, equals(0.5));
+  });
```

Comment:
please add tests for doing this that verify all the various combinations of doing this and verify that the state changes are as you would expect. For example, animateTo() while animating forward. animateTo() while animating backward. animateTo() the lowerBound. animateTo() the upper bound. And so on.